postgis/mvt.c

```diff
@@ -983,13 +1130,13 @@ LWGEOM *mvt_geom(LWGEOM *lwgeom, const GBOX *gbox, uint32_t extent, uint32_t buf
 
 	resx = width / extent;
 	resy = height / extent;
-	res = (resx < resy ? resx : resy)/2;
+	res = (resx < resy ? resx : resy) / 3;
```

Comment:
why 3?

Reply:
I was doing some testing around invalidity. It seems that I have to stop using `GEOSClipByRect` altogether because of how many things are broken with it and it's impossible to detect them all. I'll try switching to `GEOSIntersection` + handling invalid.

Reply:
I am glad you see now why I initially replaced the ClipByBox2D with Intersection globally :)

Reply:
Yup, I totally get it now.

Reply:
Will using `GEOSIntersection` cause problems because it requires valid input?  If you have to run `MakeValid` on the input that will slow things down.

Reply:
> Will using GEOSIntersection cause problems because it requires valid input?

Yes, what I'm doing now in this PR is to use GEOSIntersection and, if I get an error, run `MakeValid` in the input and retry. It's far from optimal (GEOSIntersection is already slower that GEOSClipByRect) but at least I get correct results for MVT.

Reply:
@dr-jts it causes problems as it is more-than-linear complexity. @Algunenano performed benchmarks and it is much slower on box clipping on larger polygons. Would be perfect to just make it faster in newer GEOS.

Reply:
The only nice part that has come up out of this, is that since now I'm clipping per polygon instead of full multipolygons some I'm seeing performance improvements in the tests where most geometries in the multipolygons were outside. OTOH, I get a 100% slower timings on other tests because of how slow it is validating + instersecting.

Reply:
@Komzpa Yes, would be very good to improve `GEOSClipByRect` using the quick-clip (Sutherland-Hodgson) algorithm as in new JTS code.  Will need to add a capability to clip lines and points, but that's easy compared to polygons.  And will have to decide what (if any) of the original semantics to keep (since the current GEOS clipping code seems to have some functionality which may just be side-effects and not used - e.g. converting clipped holes into lines).

It would be possible to port the code now, but only by using the buffer(0) hack to fix topology.  And it isn't 100% reliable.  So probably the changeover should wait until there is a proper topology building pipeline - which will come soon I hope.